gpytorch/lazy/lazy_tensor.py

```diff
@@ -1677,11 +1677,19 @@ def root_inv_decomposition(self, initial_vectors=None, test_vectors=None, method
         if self.shape[-2:].numel() == 1:
             return RootLazyTensor(1 / self.evaluate().sqrt())
 
-        # no need to do extra work if we already have a cached cholesky decomposition
-        # TODO: Resolve issues related to triangular tensor instantiation \
-        # if _is_in_cache_ignore_args(self, "cholesky"):
-        #     res = self.cholesky()
-        #     return CholLazyTensor(res).inverse()
+        # check to see if we have already run lanczos for a diagonalization
+        try:
+            evals, evecs = get_from_cache(self, "diagonalization")
+            F = evecs * evals.clamp(1e-7).sqrt().reciprocal().unsqueeze(-2)
```

Comment:
Equivalent but more clear what's happening

```suggestion
            F = evecs * evals.clamp_min(1e-7).sqrt().reciprocal().unsqueeze(-2)
```